runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/ExecutionStateSampler.java

```diff
@@ -36,8 +36,7 @@
 /** Monitors the execution of one or more execution threads. */
 public class ExecutionStateSampler {
 
-  private final ConcurrentSkipListSet<ExecutionStateTracker> activeTrackers =
-      new ConcurrentSkipListSet<>();
+  private final HashSet<ExecutionStateTracker> activeTrackers = new HashSet<>();
```

Comment:
`@GuardedBy(this)`? Or a comment to that effect. Minor benefit of the annotation over a comment is that there are analyses (including checkerframework that we already use) that can check them statically or dynamically.

Reply:
Thanks! I forgot about the Java annotations.